### PR TITLE
[CSL 159] Add User Segments to iOS SDK

### PR DESCRIPTION
**Pull request by jjl014**
#### Updates:
* Add support for user segments
* Add tests

#### Notes:
* Was running into issues building because it was trying to use the new build system. Switching back to legacy fixes the issues. I wonder why it doesn't work with the new build system? 🤔

### CODE REVIEW

File: AutocompleteClient/Constants/Constants.swift [hunk 1]
```diff
@@ -56,6 +56,7 @@ struct Constants {
         static let apiKey = "key"
         static let baseURLString = "https://ac.cnstrc.com"
         static let httpMethod = "GET"
+        static let segments = "us"
```
Comment (jjl014) [inline AutocompleteClient/Constants/Constants.swift:59]: Remove from constants

File: AutocompleteClient/FW/Logic/Request/Builder/RequestBuilder+QueryItems.swift [hunk 2]
```diff
@@ -124,6 +124,18 @@ extension RequestBuilder {
         queryItems.add(URLQueryItem(name: formattedKey, value: testCellValue))
     }
 
+    func set(segments: [String]?) {
```
Comment (jjl014) [inline AutocompleteClient/FW/Logic/Request/Builder/RequestBuilder+QueryItems.swift:127]: Might be moving this as well as test cells over to `RequestBuilder.swift`

File: AutocompleteClient/Constants/Constants.swift (hunk 1 above)
Comment (jjl014) [reply AutocompleteClient/Constants/Constants.swift]: Updated! 👍

File: AutocompleteClient/FW/Logic/Request/Builder/RequestBuilder+QueryItems.swift (hunk 2 above)
Comment (jjl014) [reply AutocompleteClient/FW/Logic/Request/Builder/RequestBuilder+QueryItems.swift]: Updated! 👍

**Review by crgee1 — COMMENTED** [review]
looks good, unsure about the build stuff though

**Review by rubikzube — APPROVED** [review]

**Pull request merged**